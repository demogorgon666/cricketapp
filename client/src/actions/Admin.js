import {
  GET_PLAYERS,
  CREATE_PLAYER,
  CREATE_TEAM,
  EDIT_PLAYER,
  EDIT_TEAM,
  DELETE_PLAYER,
  DELETE_TEAM,
  GET_PLAYER_SEARCH,
  GET_TEAM_SEARCH,
  GET_ALLTEAMS,
  ERROR_TYPE
} from "./Types";
import axios from "axios";

// export const getPlayers = () => dispatch => {
// 	return axios
// 		.get("http://localhost:5000/api/cricketalpha/allPlayer")
// 		.then(res => {
// 			console.log("From Player Actions", res.data.data);
// 			dispatch({
// 				type: GET_PLAYERS,
// 				payload: res.data.data
// 			});
// 		})
// 		.catch(err => {
// 			console.log(err);
// 		});
// };
//
export const getPlayers = () => dispatch => {
  console.log(localStorage.getItem("token"));
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/allPlayer", {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: GET_PLAYERS,
        payload: res.data.data
      });
      console.log(res.data);
    })
    .catch(err => {
      console.log(err);
    });
};

// export const getTeams = () => dispatch => {
//     return axios
//       .get("http://localhost:5000/api/cricketalpha/allteams")
//       .then(res => {
//         dispatch({
//           type: GET_TEAMS,
//           payload: res.data.data
//         });
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };

export const getAllTeams = () => dispatch => {
  console.log(localStorage.getItem("token"));
  return axios
    .get("http://localhost:5000/apis/admin/allteam", {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: GET_ALLTEAMS,
        payload: res.data.data
      });
      console.log(res.data);
    })
    .catch(err => {
      console.log(err);
    });
};

//   export const createPlayer = player => dispatch => {
//     return axios
//       .post("http://localhost:5000/api/cricketalpha/player/new", player)
//       .then(res => {
//         dispatch({
//           type: CREATE_PLAYER
//         });
//         console.log("User created successfully");
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };

//   export const createTeam = team => dispatch => {
//     return axios
//       .post("http://localhost:5000/api/cricketalpha/user/new", team)
//       .then(res => {
//         dispatch({
//           type: CREATE_TEAM
//         });
//         console.log("User created successfully");
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };
export const createTeam = team => dispatch => {
  return axios
    .post("http://localhost:5000/apis/admin/team/new", team, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: CREATE_TEAM
      });
      alert("Added one team successfully");
      dispatch(getAllTeams());
    })
    .catch(err => {
      console.log(err);
      dispatch({ type: ERROR_TYPE, payload: err.response.data.message });
      console.log(err.response.data.message);
    });
};

export const createPlayer = player => dispatch => {
  return axios
    .post("http://localhost:5000/apis/admin/player/new", player, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: CREATE_PLAYER
      });
      alert("Added one player successfully");
      dispatch(getPlayers());
    })
    .catch(err => {
      console.log(err);
      dispatch({ type: ERROR_TYPE, payload: err.response.data.message });
      console.log(err.response.data.message);
    });
};

export const editPlayer = player => dispatch => {
  return axios
    .put(
      "http://localhost:5000/apis/admin/editplayer/" + player.player_id,
      player,
      {
        headers: {
          Authorization: `Bearer ${localStorage.getItem("token")}`
        }
      }
    )
    .then(res => {
      dispatch({
        type: EDIT_PLAYER
      });

      alert("Edited Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const editTeam = team => dispatch => {
  return axios
    .put("http://localhost:5000/apis/admin/editteam/" + team.team_id, team, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: EDIT_TEAM
      });
      // dispatch(getUsers());
      alert("Edited Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const deletePlayer = player_id => dispatch => {
  return axios
    .delete("http://localhost:5000/apis/admin/deleteplayer/" + player_id, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: DELETE_PLAYER
      });
      dispatch(getPlayers());
      // alert("Player Deleted Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const deleteTeam = team_id => dispatch => {
  return axios
    .delete("http://localhost:5000/apis/admin/deleteteam/" + team_id, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: DELETE_TEAM
      });
      dispatch(getAllTeams());
      // alert("Team Deleted Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const getPlayerSearch = searchString => dispatch => {
  return axios
    .get("http://localhost:5000/apis/admin/search/" + searchString)
    .then(res => {
      dispatch({
        type: GET_PLAYER_SEARCH,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTeamSearch = searchString => dispatch => {
  return axios
    .get("http://localhost:5000/apis/admin/searchteam/" + searchString)
    .then(res => {
      dispatch({
        type: GET_TEAM_SEARCH,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};
