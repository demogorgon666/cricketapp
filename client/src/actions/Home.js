import {
  GET_PLAYERS,
  GET_STATISTICS,
  GET_PLAYER_STATS,
  GET_TOP_BATSMAN,
  GET_NEWS,
  GET_HOME_RECENT_MATCHES
} from './Types'
import axios from 'axios'

export const getNews = () => dispatch => {
  axios
    .get('http://localhost:5000/apis/news')
    .then(res => {
      dispatch({
        type: GET_NEWS,
        payload: res.data.data
      })
      console.log("from 'then' get news")
    })

    .catch(err => console.log(err))
}

// fetching recent matches
export const getRecentMatches = () => dispatch => {
  console.log('from actions of home recent matches')
  axios
    .get('http://localhost:5000/apis/recentMatches')
    .then(res => {
      dispatch({
        type: GET_HOME_RECENT_MATCHES,
        payload: res.data.data
      })
      console.log("from 'then' get RECENT MATCHES")
    })

    .catch(err => console.log(err))
}
