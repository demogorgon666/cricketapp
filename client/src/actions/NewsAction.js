import { GET_NEWS_ID } from "./Types";
import axios from "axios";
import { id } from "date-fns/esm/locale";

export const getNewsById = id => dispatch => {
  console.log("getnews by id", id);
  return axios
    .get("http://localhost:5000/apis/News/newsbyid/" + id)
    .then(res => {
      console.log("From newsbyid actions", res.data.data);
      dispatch({
        type: GET_NEWS_ID,
        payload: res.data.data,
      });
    })
    .catch(err => {
      console.log(err);
    });
};
