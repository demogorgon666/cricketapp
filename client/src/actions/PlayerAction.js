import {
  GET_PLAYERS,
  GET_SINGLE_PLAYER,
  GET_ALL_BATSMAN,
  GET_ALL_BOWLERS,
  GET_BATSMAN_STATS,
  GET_ODI_BATSMAN_STATS,
  GET_T20_BATSMAN_STATS,
  GET_ODI_BOWLER_STATS,
  GET_T20_BOWLER_STATS,
  GET_TEST_BOWLER_STATS,
  GET_TOP_SIXES
} from "./Types";
import axios from "axios";

export const getPlayers = () => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/allPlayer")
    .then(res => {
      console.log("From Player Actions", res.data.data);
      dispatch({
        type: GET_PLAYERS,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getSinglePlayer = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/singlePlayer/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions Single player is: ",
        res.data.data[0].player_name
      );
      dispatch({
        type: GET_SINGLE_PLAYER,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getBatsmen = match_type => dispatch => {
  return axios
    .post("http://localhost:5000/apis/PlayerInfo/TopBatsman", match_type)
    .then(res => {
      // console.log("From Player Actions batsman data is: ", res.data.data);
      dispatch({
        type: GET_ALL_BATSMAN,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getBowlers = match_type => dispatch => {
  return axios
    .post("http://localhost:5000/apis/PlayerInfo/TopBowlers", match_type)
    .then(res => {
      // console.log("From Player Actions bowler data is: ", res.data.data);
      dispatch({
        type: GET_ALL_BOWLERS,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTopSixes = match_type => dispatch => {
  return axios
    .post("http://localhost:5000/apis/PlayerInfo/TopSixes", match_type)
    .then(res => {
      // console.log("From Player Actions bowler data is: ", res.data.data);
      dispatch({
        type: GET_TOP_SIXES,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getODIBatsmanStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/ODI-Batsman-Stats/" + player_id)
    .then(res => {
      // console.log("From Player Actions odi Batsman stats is: ", res.data.ODI);
      dispatch({
        type: GET_ODI_BATSMAN_STATS,
        payload: res.data.ODI
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getODIBowlerStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/ODIBowlerStats/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions odi BOWLER stats is: ",
        res.data.ODIBowler
      );
      dispatch({
        type: GET_ODI_BOWLER_STATS,
        payload: res.data.ODIBowler
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTestBatsmanStats = player_id => dispatch => {
  return axios
    .get(
      "http://localhost:5000/apis/PlayerInfo/Test-Batsman-Stats/" + player_id
    )
    .then(res => {
      // console.log("From Player Actions test Batsman stats is: ", res.data.Test);
      dispatch({
        type: GET_BATSMAN_STATS,
        payload: res.data.Test
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTestBowlerStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/Test-Bowler-Stats/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions test Batsman stats is: ",
        res.data.TestBowler
      );
      dispatch({
        type: GET_TEST_BOWLER_STATS,
        payload: res.data.TestBowler
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getT20BatsmanStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/T20-Batsman-Stats/" + player_id)
    .then(res => {
      // console.log("From Player Actions test Batsman stats is: ", res.data.T20);
      dispatch({
        type: GET_T20_BATSMAN_STATS,
        payload: res.data.T20
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getT20BowlerStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/T20-Bowler-Stats/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions ODI BOWLER stats is: ",
        res.data.T20Bowler
      );
      dispatch({
        type: GET_T20_BOWLER_STATS,
        payload: res.data.T20Bowler
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getBatsmanStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/Batsman-Stats/" + player_id)
    .then(res => {
      console.log("From Player Actions test Batsman stats is: ", res.data.data);
      dispatch({
        type: GET_T20_BOWLER_STATS,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};
