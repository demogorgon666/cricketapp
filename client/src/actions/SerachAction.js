import { GET_SEARCH } from "./Types";
import axios from "axios";
import Search from "../reducers/Search";

export const getSearch = (search_term, history) => dispatch => {
	console.log("actions getsearch", search_term);
	return axios
		.post("http://localhost:5000/apis/Search/search", search_term)
		.then(res => {
			console.log("From Search Actions", res.data.data);
			dispatch({
				type: GET_SEARCH,
				payload: res.data.data
			});
			console.log("actions history", history);
			console.log("res", res.data.data[0].player_id);

			history.push("/playerInfo/" + res.data.data[0].player_id);
		})
		.catch(err => {
			console.log(err);
		});
};
