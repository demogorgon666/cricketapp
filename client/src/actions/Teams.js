import {
	GET_TEAMS,
	GET_RANKS,
	GET_MATCHBYTEAMID,
	GET_TEAM_BATSMEN,
	GET_TEAM_BOWLERS,
	GET_TEAM_FIXTURES,
	GET_TEAM_STATS
} from "./Types";

import { startLoading, stopLoading } from "./LoadingAction";

import axios from "axios";

export const getTeams = match_type => dispatch => {
	return axios
		.post("http://localhost:5000/cricketalpha/teams", match_type)
		.then(res => {
			dispatch({
				type: GET_TEAMS,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

export const getRanks = (gender, ranking) => dispatch => {
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/rankings/" + gender,
			ranking
		)
		.then(res => {
			dispatch({
				type: GET_RANKS,
				payload: res.data.data
			});
			console.log("from actions", ranking);
		})
		.catch(err => {
			console.log(err);
		});
};

export const getMatch = (team_id, gender, match_type) => dispatch => {
	dispatch(startLoading());
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/match/" +
				team_id +
				"/" +
				gender,
			match_type
		)
		.then(res => {
			dispatch(stopLoading());
			dispatch({
				type: GET_MATCHBYTEAMID,
				payload: res.data.data
			});
		})
		.catch(err => {
			dispatch(startLoading());
			console.log(err);
		});
};

export const getTeamBatsmen = (gender, match_type) => dispatch => {
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/topbatsmen/" + gender,
			match_type
		)
		.then(res => {
			dispatch({
				type: GET_TEAM_BATSMEN,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

export const getTeamBowlers = (gender, match_type) => dispatch => {
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/topbowlers/" + gender,
			match_type
		)
		.then(res => {
			dispatch({
				type: GET_TEAM_BOWLERS,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

//NIKHIL
export const getFixtures = team_name => dispatch => {
	console.log("get fix from actions", team_name);
	return axios
		.post("http://localhost:5000/cricketalpha/teams/fixtures", team_name)
		.then(res => {
			dispatch({
				type: GET_TEAM_FIXTURES,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

export const getTeamStats = (team_id, gender, type) => dispatch => {
	console.log("getTeamStats action- ", team_id, gender, type);

	// dispatch(startLoading());
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/stats/" +
				team_id +
				"/" +
				gender,
			type
		)

		.then(res => {
			// dispatch(stopLoading());
			console.log("getTeamStats Action");
			dispatch({
				type: GET_TEAM_STATS,
				payload: res.data.data
			});
		})
		.catch(err => {
			// dispatch(startLoading());
			console.log(err);
		});
};
