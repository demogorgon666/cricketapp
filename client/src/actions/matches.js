import { GET_RECENT_MATCHES, GET_MATCHES_DATE, GET_MATCH_DETAILS_BY_ID } from './Types'
import axios from "axios";

const url = 'http://localhost:5000/api/matches';

export const getRecentMatches = (date) => dispatch => {
    return axios
        .get(url + "/recent/" + date, {
        })
        .then(res => {
            dispatch({
                type: GET_RECENT_MATCHES,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

export const getMatchesDate = () => dispatch => {
    return axios
        .get(url + "/bydate", {
        })
        .then(res => {
            dispatch({
                type: GET_MATCHES_DATE,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

export const getmatchdetailbyId = (id) => dispatch => {
    return axios
        .get(url + "/summary/" + id, {
        })
        .then(res => {
            dispatch({
                type: GET_MATCH_DETAILS_BY_ID,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

