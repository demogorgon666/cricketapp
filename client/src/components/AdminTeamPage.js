import React, { Component } from "react";
// import { Link } from "react-router-dom";
import {
  createTeam,
  editTeam,
  deleteTeam,
  getTeamSearch,
  getAllTeams
} from "../actions/Admin";
import { connect } from "react-redux";
import "./css/Adminpage.css";
import NavBar from "./common/Navbar";

export class AdminTeamPage extends Component {
  componentDidMount() {
    if (!localStorage.getItem("token")) {
      this.props.history.push("/login");
    }
    this.props.getAllTeams();
  }
  state = {
    team_name: "",
    showError: false,
    errorMessage: "",
    searchString: ""
  };

  OnChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  onRegister = e => {
    e.preventDefault();
    if (!this.state.team_name) {
      this.setState({
        showError: true,
        errorMessage: "Enter all the fields"
      });
    } else {
      if (this.state.team_name < 2) {
        this.setState({
          showError: true,
          errorMessage: "Please enter valid team name!"
        });
      } else {
        this.setState({
          showError: false,
          errorMessage: ""
        });
        let team = {
          team_name: this.state.team_name
        };
        this.props.createTeam(team);
        //   this.props.history.push("/login");
        this.setState({
          team_name: ""
        });
      }
    }
  };
  onSearchInputChange = event => {
    if (event.target.value) {
      this.setState({ searchString: event.target.value });
      this.props.getTeamSearch(this.state.searchString);
    } else {
      this.setState({ searchString: "" });
    }
  };

  render() {
    return (
      <div>
        <NavBar />
        <div className="player-page">
          <div className="players">
            <div className="inner-heading">
              <div>
                <h1>All Team</h1>
              </div>
              <div>
                {" "}
                <input
                  type="text"
                  name="Search"
                  margin="normal"
                  placeholder="Search"
                  onChange={this.onSearchInputChange}
                />
              </div>
            </div>
            <div className="player-list">
              {this.props.team.length == 0 ? (
                <div className="loader"></div>
              ) : (
                this.props.team.map(teams => (
                  <div className="player-name">
                    <p>{teams.team_name}</p>
                    <div className="inner-button">
                      <div style={{ marginRight: "5px" }}>
                        <button
                          onChange={this.OnChange}
                          onClick={() => {
                            this.props.history.push(
                              "/admineditteam/" + teams.team_id,
                              { teams }
                            );
                          }}
                        >
                          Edit
                        </button>
                      </div>

                      <div>
                        <button
                          onChange={this.OnChange}
                          onClick={() => this.props.deleteTeam(teams.team_id)}
                          style={{ background: "#E74C3c" }}
                        >
                          Delete
                        </button>
                      </div>
                    </div>
                  </div>
                ))
              )}
            </div>
          </div>

          <div id="Adminform">
            <fieldset>
              <h1>Add New Team</h1>
              <input
                type="text"
                name="team_name"
                placeholder="Enter Team Name"
                value={this.state.team_name}
                onChange={this.OnChange}
              />

              <span
                className="errorMessage"
                style={{
                  color: "#c0392b",
                  display: this.state.showError ? "block" : "none"
                }}
              >
                {this.state.errorMessage}
              </span>
              <button onChange={this.OnChange} onClick={this.onRegister}>
                Add Team
              </button>
            </fieldset>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  team: state.AdminTeamReducer.team
});

export default connect(
  mapStateToProps,
  { getAllTeams, createTeam, editTeam, deleteTeam, getTeamSearch }
)(AdminTeamPage);
