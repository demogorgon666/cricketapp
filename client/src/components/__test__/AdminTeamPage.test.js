import React from "react";
import { shallow } from "enzyme";
import { AdminTeamPage } from "../AdminTeamPage";
// import store from "../../store";
// import { Provider } from "react-redux";
import { getTeams } from "../../actions/Admin";
const adminteampage = jest.fn();
let teams = [];

const wrapper = shallow(
  <AdminTeamPage
    adminteampage={adminteampage}
    teams={teams}
    getTeams={getTeams}
  />
);

describe("test  Component", () => {
  it("render the component", () => {
    // const wrapper = shallow(<Register register={register} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("there should be a div", () => {
    expect(wrapper.find("div").length).toBe(8);
  });
  // it("there should be a navbar", () => {
  //   expect(wrapper.find("NavBar").length).toBe(1);
  // });
  it("there should be a h1", () => {
    expect(wrapper.find("h1").length).toBe(2);
    expect(
      wrapper
        .find("h1")
        .at(0)
        .text()
    ).toBe("All Team");
    expect(
      wrapper
        .find("h1")
        .at(1)
        .text()
    ).toBe("Add New Team");
  });
  it("there should be a input", () => {
    expect(wrapper.find("input").length).toBe(2);
  });
  it("there should be a buttons", () => {
    expect(wrapper.find("button").length).toBe(1);
  });
  // it("there should be a link", () => {
  //   expect(wrapper.find("Link").length).toBe(2);
  // });

  it("there should be a fieldset tag", () => {
    expect(wrapper.find("fieldset").length).toBe(1);
  });

  // it("there should be a p tag", () => {
  //   expect(wrapper.find("p").length).toBe(1);
  //   expect(wrapper.find("p").text()).toBe("Already have an account ?Login");
  // });
  it("there should be a button", () => {
    expect(wrapper.find("button").text()).toBe("Add Team");
  });

  // it("should change the state of testClicked to true when test tab is clicked", () => {
  //   const test = wrapper.find("button").at(0);
  //   test.simulate("click");
  //   expect(wrapper.state().onRegister).toBe(true);
  // });

  it("there should be a span", () => {
    expect(wrapper.find("span").length).toBe(1);
  });

  it("there should type inputs", () => {
    expect(
      wrapper
        .find("input")
        .at(0)
        .prop("type")
    ).toEqual("text");
    expect(
      wrapper
        .find("input")
        .at(1)
        .prop("type")
    ).toEqual("text");
  });

  it("there should name in  inputs", () => {
    expect(
      wrapper
        .find("input")
        .at(0)
        .prop("name")
    ).toEqual("Search");
    expect(
      wrapper
        .find("input")
        .at(1)
        .prop("name")
    ).toEqual("team_name");
  });
  it("there should name in  inputs", () => {
    expect(
      wrapper
        .find("input")
        .at(0)
        .prop("placeholder")
    ).toEqual("Search");
    expect(
      wrapper
        .find("input")
        .at(1)
        .prop("placeholder")
    ).toEqual("Enter Team Name");
  });
});
