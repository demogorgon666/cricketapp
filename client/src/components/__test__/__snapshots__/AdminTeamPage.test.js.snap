// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test  Component render the component 1`] = `
<div>
  <div
    className="player-page"
  >
    <div
      className="players"
    >
      <div
        className="inner-heading"
      >
        <div>
          <h1
            id="heading-1"
          >
            All Team
          </h1>
        </div>
        <div>
           
          <input
            id="searchInput"
            margin="normal"
            name="Search"
            onChange={[Function]}
            placeholder="Search"
            type="text"
          />
        </div>
      </div>
      <div
        className="player-list"
      >
        <div
          className="player-name"
        >
          <p>
            India
          </p>
          <div
            className="inner-button"
          >
            <div
              style={
                Object {
                  "marginRight": "5px",
                }
              }
            >
              <button
                className="admininnerbutton"
                id="editbutton"
                onChange={[Function]}
                onClick={[Function]}
              >
                Edit
              </button>
            </div>
            <div>
              <button
                className="admininnerbutton"
                id="deletebutton"
                onChange={[Function]}
                onClick={[Function]}
                style={
                  Object {
                    "background": "#E74C3c",
                  }
                }
              >
                Delete
              </button>
              <SweetAlert
                animation={true}
                cancelButtonText="No, keep it"
                confirmButtonColor="#e74c3c"
                confirmButtonText="Delete!"
                customClass={null}
                html={false}
                imageSize="80x80"
                imageUrl={null}
                inputPlaceholder={null}
                inputType="text"
                inputValue={null}
                onCancel={[Function]}
                onConfirm={[Function]}
                onEscapeKey={[Function]}
                onOutsideClick={[Function]}
                show={false}
                showCancelButton={true}
                showConfirmButton={true}
                showLoaderOnConfirm={false}
                text="Are you sure?"
                title="DELETE TEAM"
                type="warning"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      id="Adminform"
    >
      <fieldset>
        <h2
          id="heading-2"
        >
          Add New Team
        </h2>
        <input
          id="team_name"
          name="team_name"
          onChange={[Function]}
          placeholder="Enter Team Name"
          type="text"
          value=""
        />
        <div
          className="errorMessage"
          style={
            Object {
              "color": "#c0392b",
            }
          }
        >
          <span
            className="errorMessage"
            style={
              Object {
                "color": "#c0392b",
                "display": "none",
              }
            }
          />
        </div>
        <button
          className="adminpagebutton"
          id="add-team"
          onChange={[Function]}
          onClick={[Function]}
        >
          Add Team
        </button>
      </fieldset>
    </div>
  </div>
</div>
`;
